test/server/connection_handler_test.cc

```diff
@@ -263,6 +267,9 @@ TEST_F(ConnectionHandlerTest, RemoveListenerDuringRebalance) {
   EXPECT_CALL(dispatcher_, post(_)).WillOnce(SaveArg<0>(&post_cb));
   Network::MockConnectionSocket* connection = new NiceMock<Network::MockConnectionSocket>();
   current_handler->incNumConnections();
+#ifndef NDEBUG
```

Comment:
Why is the `#ifndef` needed?

Reply:
There is #ifndef defined just below this. Honestly, I don't know why or why we have two variants of this test case.

Reply:
Got it. Very strange, but the comment below, and linked comment in ~ActiveTcpListener, explain what's going on. This seems like it's not making anything worse then; just maintaining status quo.